lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java

```diff
@@ -223,17 +229,35 @@ private void generateCode(MethodDefinition methodDefinition) throws IOException
       out.println("  }");
 
       out.println();
+      if (!methodDefinition.needsFrame()) {
+        out.println("  @Override");
+        out.println("  public final InlinedCallNode createDirectCallNode() {");
+        out.println("    var n = " + methodDefinition.getConstructorExpression() + ";");
+        out.println("    return new InlinedCallNode<>(new Internals(internals.staticOfInstanceMethod), n) {");
+        out.println("      public Object call(Object[] args) {");
+        out.println("        return handleExecute(extra, body, args);");
+        out.println("      }");
+        out.println("    };");
+        out.println("  }");
+      }
 
       out.println("  @Override");
       out.println("  public Object execute(VirtualFrame frame) {");
-      out.println("    State state = Function.ArgumentsHelper.getState(frame.getArguments());");
+      if (methodDefinition.needsFrame()) {
+        out.println("    var args = frame.getArguments();");
+      } else {
+        out.println("    return handleExecute(this.internals, bodyNode, frame.getArguments());");
```

Comment:
What about renaming to executeInline? Isn't that more clear? `handleExecute` is too cryptic IMHO.

Reply:
There is a method `RootNode.execute` and it delegates to a method which handles all such execution - the name of the method is `handleExecute`. The same method is used from `InlinedCallNode.call` to force Truffle to inline `handleExecute`. However in case of `BuiltinRootNode.execute` there is no inlining enforced. As such I am afraid `executeInline` name would be misleading.